lib/action_view/component/content_areas.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+module ActionView
+  module Component # :nodoc:
+    module ContentAreas
+      extend ActiveSupport::Concern
+
+      included do
+        class_attribute :content_areas, default: []
+        self.content_areas = [] # default doesn't work until Rails 5.2
+      end
+
+      class_methods do
+        def set_content_areas(*areas)
```

Comment:
would. `self.content_areas = [:title, :footer]` be a more idiomatic way?

Reply:
Can you add documentation on how this method would be used, either inline or in the README?

Reply:
Yup documentation upcoming.

Reply:
added